### ISSUE TITLE
torch.compile and check pytorch nightly

### ISSUE THREAD

**Issue by MJK12341234** [KempnerInstitute/tmrc#59]
Suggestions from today's session with Horace - remove compile flag (we always want to torch.compile flex attention) and call torch.compile on model in train script. Optionally, check pytorch nightly to see if we can compile block mask creation in a better way

**Comment by timothyngo** [comment]
Let's reduce the scope of this issue to just removing the compile flag for flex attention. The other things in this issue can be re-raised in new issues.
